app/access/models.py

```diff
@@ -0,0 +1,37 @@
+from typing import Iterable
+
+from django.core.validators import validate_email
+from django.db import models
+from django.db.models.base import ModelBase
+from django.utils.translation import pgettext_lazy as _
+
+
+class User(models.Model):
+
+    _context = "User model"
+
+    def __str__(self) -> str:
+        return str(self.first_name) + str(self.last_name)
+
+    username = models.CharField(_(_context, "User name"), primary_key=True)
+    first_name = models.CharField(_(_context, "First name"))
+    last_name = models.CharField(_(_context, "Last name"))
+    email = models.CharField(_(_context, "Email"))
```

Comment:
Why not using the [EmailField](https://docs.djangoproject.com/en/5.1/ref/models/fields/#emailfield), this should use the `EmailValidator`, no?

Reply:
Because it would duplicate what is done in `Model.clean()`: [`validate_email()`](https://docs.djangoproject.com/en/5.1/ref/validators/#validate-email) is just a wrapper for `EmailValidator`.

There would be no harm in this at the moment but having it in two places risks having it out of sync at some point.

No strong opinion though, would you prefer to have it in both places?

Reply:
If I understand correctly, django will call the validator of an email field itself when doing a full clean, so overwriting clean wouldn't be necessary. But not 100% sure about this.

Reply:
I checked and you are right! As `Model.save()` now runs `Model.full_clean()`, it also runs the validators attached to the model fields. Thanks, will fix!